### DISCUSSION TITLE
how to disable tab if isNewOrDeleted

### DISCUSSION THREAD

**Discussion by arunkumar12712** [serenity-is/Serenity#7226] [General]
i am try this but not working 
TabsExtensions.setDisabled(this.tabs, "test", this.isNewOrDeleted())
<img width="409" alt="image" src="https://github.com/user-attachments/assets/e4353d93-23a4-4eb1-9dd0-74bb4124d459">

**Comment by LSNicholls** [comment]
When I use TabsExtensions, I have used an index value, not the "name" of the tab as a reference, something like this:

   ```
const projectTabInfo  = { "Customer": 0, "Project": 1, "Proposal": 2, "Execution": 3, "Reports": 4 };
   //...
   this.tabSet = document.getElementsByClassName("property-tabs")[0] as HTMLElement;
   //...  
  TabsExtensions.selectTab(this.tabSet, projectTabInfo["Project"]);

```

I just tried it, and it worked fine for setDisabled as well.  So, it's possible that you are using the wrong tab key value as your second argument?

Alternatively, maybe it is the timing of when you are trying to do the disabling; if it is too early, this.isNewOrDeleted() might not be working.  You should do it in **afterLoadEntity** or later.
